components/accept-financial-contributions/AcceptContributionsMyself.js

```diff
@@ -0,0 +1,393 @@
+                            </Flex>
+                          )}
+
+                          <Flex justifyContent={'center'} mb={4}>
+                            <StyledButton
+                              fontSize="13px"
+                              minWidth={'85px'}
+                              minHeight="36px"
+                              type="button"
+                              onClick={() => {
+                                Router.pushRoute('accept-financial-contributions', {
+                                  slug: this.props.collective.slug,
+                                  path: this.props.router.query.path,
+                                }).then(() => window.scrollTo(0, 0));
+                              }}
+                            >
+                              {intl.formatMessage(this.messages.cancel)}
+                            </StyledButton>
+                            <StyledButton
+                              fontSize="13px"
+                              minWidth={'85px'}
+                              minHeight="36px"
+                              ml={2}
+                              buttonStyle="dark"
+                              type="submit"
+                              loading={this.state.loading}
+                              onSubmit={handleSubmit}
+                            >
+                              {intl.formatMessage(this.messages.save)}
+                            </StyledButton>
+                          </Flex>
+                        </Form>
+                      );
+                    }}
+                  </Formik>
+                </Box>
+              </Flex>
+
+              <FinancialContributionsFAQ />
+            </Box>
+          )}
+          {!router.query.method && (
+            <Flex flexDirection="column" justifyContent="center" alignItems="center" my={[5]}>
+              <Box alignItems="center">
+                <Flex justifyContent="center" alignItems="center" flexDirection={['column', 'row']}>
+                  <Container alignItems="center" width={[null, 280, 312]} mb={[2, 0]}>
+                    <Flex flexDirection="column" justifyContent="center" alignItems="center">
+                      <Image src={stripeIllustration} alt="" />
+                      <StyledButton
+                        fontSize="13px"
+                        buttonStyle="dark"
+                        minHeight="36px"
+                        mt={[2, 3]}
+                        mb={3}
+                        minWidth={'145px'}
+                        onClick={() => this.connectStripe()}
+                      >
+                        {intl.formatMessage(this.messages.connectStripe)}
+                      </StyledButton>
+                      <Box minHeight={50} px={3}>
+                        <P color="black.600" textAlign="center" mt={[2, 3]} fontSize={['Caption', 'Paragraph']}>
+                          {intl.formatMessage(this.messages.transactionFee)}
```

Comment:
`<FormattedMessage/>` would look more natural here and in similar cases?

Reply:
Can we decide if that's a style choice we wil enforce everywhere? Because I've used intl.formatMessage on other buttons, and thought that was the preferred way.

Reply:
True, I reviewed a lot of code in recent days and I saw that pattern a lot. Let's ask @Betree and clarify a rule.

My assumption is that everything that can be a <FormattedMessage/> should be one.

Reply:
`formatMessage` is more practical, and sometimes mandatory to handle dynamic translations or when you need to output a string instead of a react component (ie. for an HTML `select`). It can also be useful to optimize the rendering of long lists. So we can't get rid of it.

We _could_ get rid of `FormattedMessage`, but it's very handy and doing so would impact productivity negatively (at least for me).

I have no preference for one or another when reading the code, even when mixed I still find both usages pretty clear. I do have a preference for `FormattedMessage` when developing because it's so easy to use (especially with the [right snippet](https://docs.opencollective.com/help/contributing/development/translations#formattedmessage)).

I don't think that we need to enforce a rule here, but if the rest of the team think that we should then I would vote for using `formatMessage` only when dealing with dynamic terms, when we need to output a string or to optimize lists rendering.

Reply:
What I personally like about `FormattedMessage` is that it's self contained, only one tag and nothing else to worry about it.

Using `formatMessage` forces you to first define the messages, then use it, in 2 separated places. Also, our current practice of requiring one `key` and one `id` makes it even more verbose. Sometime, the `key` and the `id` are not matching, which can lead to confusion (see example).

```
      'description.label': {
        id: 'Fields.description',
        defaultMessage: 'Description',
      },
```

I understand that `formatMessage` is useful in some situations, but I think it should generally be the exception, not the rule.

Reply:
I agree @znarf `<FormattedMessage>` allows you to use one id in multiple files and that's good.

I'm happy to go ahead and change most of my internationalised stuff to `<FormattedMessage>` if we feel we're leaning in that direction.